strutil/env.go

```diff
@@ -0,0 +1,253 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+// RawEnvironment is a slice of key=value strings.
+//
+// Using this type is required for low-level interactions, like invoking
+// execve. In other cases using Environment is both sufficient and safer.
+type RawEnvironment []string
```

Comment:
this type is used only by tests, it's probably not very useful.  the RawEnviroment on Enviroment could just be

ForExec() []string

Reply:
Done